### PR TITLE
Add Ability to Change max-wasm-size via Governance

### LINKED ISSUE #62: ✨ Parameterize max Wasm size in wasm-storage module
## ✨ Feature

Currently, the maximum Wasm size is set using a constant (~800kb). Using a module parameter for this value instead would allow us to adjust it through the governance process.

### PR DESCRIPTION
**Pull request by jim380**
## Overview
This PR attempts to achieve the followings:
- The option to override `max-wasm-size` at compile time
- The to change `max-wasm-size` via governance

Closes #62

### CODE REVIEW

File: x/wasm-storage/keeper/msg_server.go [hunk 1]
```diff
@@ -173,3 +173,16 @@ func unzipWasm(wasm []byte) ([]byte, error) {
 	}
 	return unzipped, nil
 }
+
+func (m msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if m.GetAuthority() != req.Authority {
+		return nil, fmt.Errorf("invalid authority; expected %s, got %s", m.GetAuthority(), req.Authority)
+	}
+
```
Comment (hacheigriega) [inline x/wasm-storage/keeper/msg_server.go:181]: v0.50 recommends that we validate message here

File: x/wasm-storage/keeper/msg_server.go [hunk 2]
```diff
@@ -173,3 +173,16 @@ func unzipWasm(wasm []byte) ([]byte, error) {
...
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := m.SetParams(ctx, req.Params); err != nil {
```
Comment (hacheigriega) [inline x/wasm-storage/keeper/msg_server.go:183]: validate params before setting?

File: x/wasm-storage/types/tx.go [hunk 3]
```diff
@@ -87,3 +87,15 @@ func (msg MsgInstantiateAndRegisterProxyContract) ValidateBasic() error {
 	}
 	return nil
 }
+
+func (m *MsgUpdateParams) ValidateBasic() error {
```
Comment (hacheigriega) [inline x/wasm-storage/types/tx.go:91]: maybe call it `Validate()` to avoid confusion? (since `ValidateBasic()` has been deprecated)

**Comment by hacheigriega** [comment]
Closes #62, right?

What's the idea behind the compile time parameter? Just wondering why we would ever want that when the max wasm size is a chain parameter.

File: app/app.go [hunk 4]
```diff
@@ -589,6 +590,12 @@ func NewApp(
 		panic(err)
 	}
 
+	if maxSize := os.Getenv("MAX_WASM_SIZE"); maxSize != "" {
```
Comment (hacheigriega) [inline app/app.go:593]: What's the idea behind the compile time parameter? Just wondering why we would ever want that when the max wasm size is a chain parameter.

Comment (hacheigriega) [reply app/app.go:593]: Perhaps just for testing purpose?

Comment (jim380) [reply app/app.go:593]: Because they might need to be used by `ValidateBasic()` or `Validate()` which is stateless.

Comment (jim380) [reply app/app.go:593]: But yeah if it was changed via governance afterwards, the binary wouldn't be able to pick that up without accessing state. For testing purposes for sure.

File: x/wasm-storage/keeper/msg_server.go (hunk 2 above)
Comment (jim380) [reply x/wasm-storage/keeper/msg_server.go:183]: The msg (`MsgUpdateParams`) would've been validated in `Validate()`:
```golang
func (m *MsgUpdateParams) Validate() error {
	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
		return fmt.Errorf("invalid authority address: %s", err)
	}

	if err := m.Params.ValidateBasic(); err != nil {
		return err
	}

	return nil
}
```

**Review by hacheigriega — APPROVED** [review]

**Pull request merged**